src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -52,14 +52,15 @@ class TransportManagementService implements Serializable {
 
         def response = sendApiRequest(parameters)
 
-        if (config.verbose) {
-            echo("Received response with status ${response.status} from authentication request.")
+        echo("Received response with status ${response.status} from authentication request.")
+
+        if (response.status != 200) {
+            def errorMessage = "OAuth Token retrieval failed (HTTP status code '${response.status}')."
+            signalAboutError(response, errorMessage)
```

Comment:
`signalAboutError` does never return normally, isn't it? It throws always an excpetion, right?

If that is true I would like to suggest to adjust the name according so that we can see that from the method name (e.g. prepareAndThrowException, raiseError/Exception or something similar).

In that case the `else` below can be dropped (not the coding executed in the body of the else block, of course) since we can never reach it when the if statement evaluates to `true`.

Reason for that comment: with the current style there is no clearly defined return value for the cases we don't end up in the `else`. In that case the returnValue from `signalAboutError` would be the return value and that seems to be rather blurry to me.

Reply:
Done.